x-pack/plugins/security_solution/public/cases/components/create/sync_alerts_toggle.test.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import { act, waitFor } from '@testing-library/react';
+
+import { useForm, Form, FormHook } from '../../../shared_imports';
+import { SyncAlertsToggle } from './sync_alerts_toggle';
+import { schema, FormProps } from './schema';
+
+describe('SyncAlertsToggle', () => {
+  let globalForm: FormHook;
+
+  const MockHookWrapperComponent: React.FC = ({ children }) => {
+    const { form } = useForm<FormProps>({
+      defaultValue: { syncAlerts: true },
+      schema: {
+        syncAlerts: schema.syncAlerts,
+      },
+    });
+
+    globalForm = form;
+
+    return <Form form={form}>{children}</Form>;
+  };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('it renders', async () => {
+    const wrapper = mount(
+      <MockHookWrapperComponent>
+        <SyncAlertsToggle isLoading={false} />
+      </MockHookWrapperComponent>
+    );
+
+    expect(wrapper.find(`[data-test-subj="caseSyncAlerts"]`).exists()).toBeTruthy();
+  });
+
+  it('it toggles the switch', async () => {
+    const wrapper = mount(
+      <MockHookWrapperComponent>
+        <SyncAlertsToggle isLoading={false} />
+      </MockHookWrapperComponent>
+    );
+
+    act(() => {
+      wrapper.find('[data-test-subj="caseSyncAlerts"] button').first().simulate('click');
+      wrapper.update();
+    });
+
+    await waitFor(() => {
+      expect(globalForm.getFormData()).toEqual({ syncAlerts: false });
+    });
+  });
+
+  it('it shows the correct labels', async () => {
+    const wrapper = mount(
+      <MockHookWrapperComponent>
+        <SyncAlertsToggle isLoading={false} />
+      </MockHookWrapperComponent>
+    );
+
+    await waitFor(() => {
```

Comment:
no need to wrap this with `waitFor`

Reply:
Removing `waitFor` produces this warning:

```
console.error
    Warning: An update to MockHookWrapperComponent inside a test was not wrapped in act(...).

    When testing, code that causes React state updates should be wrapped into act(...):

    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */

    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
        in MockHookWrapperComponent (created by WrapperComponent)
        in WrapperComponent

      211 |           : undefined;
      212 |
    > 213 |       setIsValid(isFormValid);
          |       ^
      214 |
      215 |       return { areFieldsValid, isFormValid };
      216 |     },

      at warningWithoutStack (node_modules/react-dom/cjs/react-dom.development.js:530:32)
      at warnIfNotCurrentlyActingUpdatesInDEV (node_modules/react-dom/cjs/react-dom.development.js:25848:7)
      at setIsValid (node_modules/react-dom/cjs/react-dom.development.js:17072:9)
      at src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_form.ts:213:7
      at src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field.ts:151:5
```

Reply:
maybe the one on L74, but it passes for me without warning when i remove the one here